<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Searching_Application</class>
 <widget class="QMainWindow" name="Searching_Application">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>690</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Searching_Application</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>0</y>
      <width>331</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>11</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Searching Visualization Application</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignmentFlag::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>50</y>
      <width>171</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Sample Visualization</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ViewLinearSearchButton">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>80</y>
      <width>111</width>
      <height>29</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Linear search</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ViewBinarySearchButton">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>80</y>
      <width>111</width>
      <height>29</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Binary search</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ViewJumpSearchButton">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>80</y>
      <width>111</width>
      <height>29</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Jump search</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ViewTernarySearchButton">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>80</y>
      <width>111</width>
      <height>29</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Ternary search</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ViewExponentialSearchButton">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>120</y>
      <width>161</width>
      <height>29</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Exponential search</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ViewInterpolationSearchButton">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>120</y>
      <width>161</width>
      <height>29</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Interpolation search</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ViewFibonacciSearchButton">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>120</y>
      <width>161</width>
      <height>29</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Fibonacci search</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>160</y>
      <width>171</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Try it yourself</string>
    </property>
   </widget>
   <widget class="QScrollArea" name="scrollArea">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>190</y>
      <width>761</width>
      <height>85</height>
     </rect>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>759</width>
       <height>83</height>
      </rect>
     </property>
     <widget class="QTextEdit" name="inputArrayEdit">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>761</width>
        <height>85</height>
       </rect>
      </property>
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QPushButton" name="UploadButton">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>280</y>
      <width>121</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Upload file</string>
    </property>
   </widget>
   <widget class="QPushButton" name="GenerateInputButton">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>320</y>
      <width>121</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Auto-generate</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>320</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Size:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="inputSizeEdit">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>320</y>
      <width>71</width>
      <height>28</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>320</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Type:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="generateTypeBox">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>320</y>
      <width>101</width>
      <height>28</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Ascending</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Descending</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Random</string>
     </property>
    </item>
   </widget>
   <widget class="QLineEdit" name="lineEdit_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>370</y>
      <width>121</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Select algorithm</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignmentFlag::AlignCenter</set>
    </property>
   </widget>
   <widget class="QCheckBox" name="linearSearchCheck">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>370</y>
      <width>121</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Linear Search</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="binarySearchCheck">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>370</y>
      <width>121</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Binary Search</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="jumpSearchCheck">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>370</y>
      <width>121</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Jump Search</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="ternarySearchCheck">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>370</y>
      <width>121</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Ternary Search</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="exponentialSearchCheck">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>410</y>
      <width>171</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Exponential Search</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="interpolationSearchCheck">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>410</y>
      <width>171</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Interpolation Search</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="fibonacciSearchCheck">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>410</y>
      <width>171</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Fibonacci Search</string>
    </property>
   </widget>
   <widget class="QScrollArea" name="scrollArea_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>480</y>
      <width>761</width>
      <height>151</height>
     </rect>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents_2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>759</width>
       <height>149</height>
      </rect>
     </property>
     <widget class="QTextEdit" name="resultLog">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>761</width>
        <height>151</height>
       </rect>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QPushButton" name="Search">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>440</y>
      <width>121</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Search!!</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="inputIntoFieldCheck">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>160</y>
      <width>161</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Field Display</string>
    </property>
   </widget>
   <widget class="QPushButton" name="InputArrayButton">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>320</y>
      <width>161</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Input this</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="searchValueEdit">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>440</y>
      <width>321</width>
      <height>28</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>440</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Search Values</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
